src/AddressRegistry/StreetName/StreetName.cs

```diff
@@ -221,18 +221,67 @@ public void CorrectStreetNameRetirement()
                     sourceAddress.Geometry.Geometry
                 );
 
-                proposedAddresses.Add(newPersistentLocalId);
-                executionContext.AddressesAdded.Add((PersistentLocalId, newPersistentLocalId));
+                proposedAddresses.Add(houseNumberPersistentLocalId);
+                executionContext.AddressesAdded.Add((PersistentLocalId, houseNumberPersistentLocalId));
+                executionContext.AddressesUpdated.Add((PersistentLocalId, houseNumberPersistentLocalId));
 
                 readdressAddresses.Add(new ReaddressAddressData(
                     item.SourceAddressPersistentLocalId,
-                    newPersistentLocalId,
+                    houseNumberPersistentLocalId,
                     sourceAddress.Status,
                     item.DestinationHouseNumber,
+                    boxNumber: null,
                     sourceAddress.PostalCode,
                     sourceAddress.Geometry,
-                    sourceAddress.IsOfficiallyAssigned
+                    sourceAddress.IsOfficiallyAssigned,
+                    parentAddressPersistentLocalId: null
                 ));
+
+                foreach (var boxNumberAddress in sourceAddress.Children)
+                {
+                    if (!boxNumberAddress.IsActive)
+                    {
+                        continue;
+                    }
```

Comment:
You can include this in the foreach? `sourceAddress.Children.Where(x => x.IsActive)`?